packages/amplication-server/src/core/prismaSchemaUtils/prismaSchemaUtils.service.ts

```diff
@@ -0,0 +1,1487 @@
+          this.validateModelNamesReservedWords(model.name);
+        if (invalidModelNamesReservedWordsErrors) {
+          errors.push(...invalidModelNamesReservedWordsErrors);
+        }
+
+        const invalidFieldNamesReservedWordsErrors =
+          this.validateFieldNamesReservedWords(field.name);
+        if (invalidFieldNamesReservedWordsErrors) {
+          errors.push(...invalidFieldNamesReservedWordsErrors);
+        }
+      });
+    });
+
+    return errors.length > 0 ? errors : null;
+  }
+
+  private validateModelName(modelName: string): ErrorMessage[] | null {
+    const errors: ErrorMessage[] = [];
+    const modelNameRegex = /^[A-Za-z][A-Za-z0-9_]*$/;
+    if (!modelNameRegex.test(modelName)) {
+      errors.push({
+        message: ErrorMessages.InvalidModelName,
+        level: ErrorLevel.Error,
+        details: `ModelName: "${modelName}" must adhere to the following regular expression: [A-Za-z][A-Za-z0-9_]*`,
+      });
+      this.logger.error(
+        `Model name "${modelName}" must adhere to the following regular expression: [A-Za-z][A-Za-z0-9_]*`,
+        null,
+        PrismaSchemaUtilsService.name
+      );
+
+      return errors.length > 0 ? errors : null;
+    }
+  }
+
+  private validateFieldName(modelName: string): ErrorMessage[] | null {
+    const errors: ErrorMessage[] = [];
+    const modelNameRegex = /^[A-Za-z][A-Za-z0-9_]*$/;
+    if (!modelNameRegex.test(modelName)) {
+      errors.push({
+        message: ErrorMessages.InvalidFieldName,
+        level: ErrorLevel.Error,
+        details: `modelName: "${modelName}" must adhere to the following regular expression: [A-Za-z][A-Za-z0-9_]*`,
+      });
+      this.logger.error(
+        `Model name "${modelName}" must adhere to the following regular expression: [A-Za-z][A-Za-z0-9_]*`,
+        null,
+        PrismaSchemaUtilsService.name
+      );
+
+      return errors.length > 0 ? errors : null;
+    }
+  }
+
+  private validateFKFieldName(
+    modelName: string,
+    fieldName: string
+  ): ErrorMessage[] | null {
+    const errors: ErrorMessage[] = [];
+    const isValidFkFieldName = isCamelCaseWithIdSuffix(fieldName);
+
+    if (!isValidFkFieldName) {
+      errors.push({
+        message: ErrorMessages.InvalidFKFieldName,
+        level: ErrorLevel.Error,
+        details: `Field name: "${fieldName}" in model: "${modelName}" must be in camelCase and end with "Id"`,
+      });
+    }
+
+    return errors.length > 0 ? errors : null;
+  }
+
+  private validateModelNamesReservedWords(
+    modelName: string
+  ): ErrorMessage[] | null {
+    const errors: ErrorMessage[] = [];
+    const isReservedModelName = isReservedName(modelName.toLowerCase().trim());
+    if (isReservedModelName) {
+      errors.push({
+        message: ErrorMessages.ReservedWord,
+        level: ErrorLevel.Warning,
+        details: `Model name: "${modelName}" is a reserved word. Please be aware that we renamed it to "${modelName}Model"`,
+      });
+    }
```

Comment:
instead of duplicate validation, use the other "formatModelName" and write the message from there